rotkehlchen/history/price.py

```diff
@@ -351,3 +353,33 @@ def query_multiple_prices(
         )
 
         return assets_price
+
+    @staticmethod
+    def query_uniswap_position_price(
+            pool_token: EvmToken,
+            pool_token_amount: FVal,
+            timestamp: Timestamp,
+    ) -> Price:
+        """Return the uniswap position price at the given timestamp"""
+        evm_inquirer = Inquirer.get_evm_manager(chain_id=pool_token.chain_id).node_inquirer
+        block_number = evm_inquirer.get_blocknumber_by_time(timestamp)
+        if pool_token.protocol == UNISWAP_PROTOCOL:
+            if (pool_price := lp_price_from_uniswaplike_pool_contract(
+                evm_inquirer=evm_inquirer,
+                token=pool_token,
+                token_price_func=PriceHistorian().query_historical_price,
+                token_price_func_args=[A_USD, timestamp],
+                block_identifier=block_number,
+            )) is not None:
+                return Price(pool_token_amount * pool_price)
+            return ZERO_PRICE
+        elif pool_token.protocol == UNISWAPV3_PROTOCOL:
+            return get_uniswap_v3_position_price(
+                token=pool_token,
+                evm_inquirer=evm_inquirer,
+                block_identifier=block_number,
+                usd_price_func=PriceHistorian().query_historical_price,
+                usd_price_func_args=[A_USD, timestamp],
+            )
+        else:
+            raise ValueError(f'Protocol {pool_token.protocol} is not supported for price queries. Only Uniswap V2 and V3 are supported.')  # noqa: E501
```

Comment:
we don't raise ValueError ever since it is too generic. Use `InputError`. Any possible exception should be documented in the docstring and also where are you handling the errors that this function can raise?